### PR TITLE
Use prebuild for binaries

### PR DESCRIPTION
**Pull request by mischnic**
The CI job should call the `prebuild` script to build+package and then the `prebuild-upload` script to upload attach the binary to a GitHub release (https://github.com/prebuild/prebuild#uploading). (I have no idea how to tell Azure that)

#10

- I guess `yarn install` on CI should always `--build-from-source` ?

### CODE REVIEW

**Comment by devongovett** [comment]
hmm, weird. re-ran and it passed.

**Comment by devongovett** [comment]
Interesting. I kind of assumed it would just publish the pre-built binaries as part of the npm package but that seems not to be the case. Instead, it downloads it dynamically at install time. Wonder why...

**Comment by mischnic** [comment]
> Interesting. I kind of assumed it would just publish the pre-built binaries as part of the npm package but that seems not to be the case. Instead, it downloads it dynamically at install time. Wonder why...

Publishing to S3 or GItHub is the common way to do this, though there are libraries to publish to npm as well: https://github.com/prebuild/prebuildify

Advantage of publishing *every* prebuilt binary in the npm tarball:
- install speed
- would also work if user disabled npm scripts

**Major** Disadvantage:
- Size

**Comment by devongovett** [comment]
Yeah makes sense. I'm close to getting things to upload to GitHub via azure pipelines.

File: azure-pipelines-template.yml [hunk 1]
```diff
@@ -41,6 +43,16 @@ jobs:
       displayName: Install Watchman
 
   - script: npm install
```
Comment (mischnic) [inline azure-pipelines-template.yml]: Should be `npm install --build-from-source` to ensure the CI doesn't download a prebuilt binary

Comment (devongovett) [reply azure-pipelines-template.yml]: ah good catch, thanks. added on master.

**Pull request merged**